src/step.ts

```diff
@@ -10,11 +10,16 @@ import { HatchetClient } from './clients/hatchet-client';
 import WorkflowRunRef from './util/workflow-run-ref';
 import { Worker } from './clients/worker';
 import { WorkerLabels } from './clients/dispatcher/dispatcher-client';
-import { WorkerLabelComparator } from './protoc/workflows';
+import { CreateStepRateLimit, RateLimitDuration, WorkerLabelComparator } from './protoc/workflows';
 
 export const CreateRateLimitSchema = z.object({
-  key: z.string(),
-  units: z.number().min(1),
+  key: z.string().optional(),
+  staticKey: z.string().optional(),
+  dynamicKey: z.string().optional(),
+
+  units: z.union([z.number().min(1), z.string()]),
```

Comment:
```suggestion
  units: z.union([z.number().min(0), z.string()]),
```

Might there be cases where a user wants to set 0 units for testing/dynamically within the runtime?